content/master/concepts/compositions.md

```diff
@@ -1308,47 +1308,69 @@ parameters in the Composition.
+`connectionDetails` are valid according to the involved resources' schemas, for
+example checking that the source/destination field of a patch is a valid field
+according to the source/destination resource's schema and that types match, if
+defined, taking into account transforms.
+
+{{<hint "note" >}}
+Composition schema aware validation is a beta feature, so Crossplane
+enables it by default. You can disable it by setting the
+`--enable-composition-webhook-schema-validation=false` flag on the Crossplane
+pod.
 The [Crossplane Pods]({{<ref "./pods#edit-the-deployment">}}) page has
 more information on enabling Crossplane flags.
+{{< /hint >}}
+
+#### Schema aware validation modes
+
+Integrity validation errors always result in a rejection of the Composition.
+
+You can configure how Crossplane should handle both missing involved resources'
+schemas and actual schema aware validation errors.
 
 {{<hint "note" >}}
-<!-- vale write-good.TooWordy = NO -->
-Schema validation only checks the `apiVersion` and `kind` are valid. Schema
-validation doesn't validate the fields of a specific resource.
-<!-- vale write-good.TooWordy = YES -->
+In case of any involved resource schema missing, Crossplane skips schema aware
+validation altogether.
 {{< /hint >}}
 
-The default validations are still checked with schema validation enabled. 
-#### Validation modes
+The following modes are available:
 
-Schema validation supports two modes:
-* `loose` (default) - Sends an warning for each schema error and installs the 
-  Composition if the default validations pass. 
-* `strict` - Send an error for every schema validation error and rejects the
-  Composition. 
+| Mode     | Missing Schema | Schema Aware Error | Integrity Error |
+| -------- | -------------- |--------------------|-----------------|
+| `warn`   | Warning        | Warning            | Error           |
+| `loose`  | Warning        | Error              | Error           |
+| `strict` | Error          | Error              | Error           |
 
 Change the validation mode for a Composition with the
-{{<hover label="mode" line="5">}}crossplane.io/composition-validation-mode{{</hover>}} 
-annotation. 
+{{<hover label="mode" line="5">}}crossplane.io/composition-schema-aware-validation-mode{{</hover>}} 
+annotation.
+
+If not specified, the default mode is `warn`.
 
-For example, to enable `strict` mode checking:
+For example, to enable `loose` mode checking:
 
 ```yaml {copy-lines="none",label="mode"}
 apiVersion: apiextensions.crossplane.io/v1
 kind: Composition
 metadata:
   annotations:
-    crossplane.io/composition-validation-mode: strict
+    crossplane.io/composition-schema-aware-validation-mode: loose
   # Removed for brevity
 spec:
   # Removed for brevity
 ```
+
+{{<hint "important" >}}
+Modes also apply to Compositions defined by Configuration packages, look at
+Crossplane's logs for warnings in case of issues. The default is just to warn,
```

Comment:
Do you mean `kubectl logs` or the messages in `kubectl describe` of something?

Reply:
Logs logs, as in kubectl logs, describe ones are events, we log warnings to stdout if we get any back for one of the calls Crossplane does. E.g. the infamous "ControllerConfig has been deprecated" which showed up in the logs